extensions/notebook/src/dialog/addNotebookDialog.ts

```diff
@@ -0,0 +1,179 @@
+		this.bookNames.unshift('-');
+	}
+
+	protected createHorizontalContainer(view: azdata.ModelView, items: azdata.Component[]): azdata.FlexContainer {
+		return view.modelBuilder.flexContainer().withItems(items, { CSSStyles: { 'margin-right': '5px', 'margin-bottom': '10px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
+	}
+
+	public async selectFolder(): Promise<string | undefined> {
+		const allFilesFilter = loc.allFiles;
+		let filter: any = {};
+		filter[allFilesFilter] = '*';
+		let uris = await vscode.window.showOpenDialog({
+			filters: filter,
+			canSelectFiles: false,
+			canSelectMany: false,
+			canSelectFolders: true,
+			openLabel: loc.labelSelectFolder
+		});
+		if (uris && uris.length > 0) {
+			return uris[0].fsPath;
+		}
+		return undefined;
+	}
+
+	public async validatePath(folderPath: string, fileBasename: string): Promise<boolean> {
+		const destinationUri = path.join(folderPath, fileBasename);
+		if (await pathExists(destinationUri)) {
+			const doReplace = await confirmReplace(this.prompter, loc.confirmReplaceFile);
+			if (doReplace) {
+				//remove folder if exists
+				await remove(destinationUri);
+				return true;
+			}
+			return false;
+		}
+		return await pathExists(folderPath);
+	}
+
+	public async createDialog(): Promise<void> {
+		this.dialog = azdata.window.createModelViewDialog(loc.newNotebook);
+		this.dialog.registerContent(async view => {
+			this.view = view;
+
+			const notebookLabel = this.view.modelBuilder.text()
+				.withProperties({
+					value: loc.notebookDescription,
+					CSSStyles: { 'margin-bottom': '0px', 'margin-top': '0px', 'font-size': 'small' }
+				}).component();
+
+			this.notebookNameInputBox = this.view.modelBuilder.inputBox()
+				.withProperties({
+					values: [],
+					value: '',
+					enabled: true
+				}).component();
+
+			this.saveLocationInputBox = this.view.modelBuilder.inputBox().withProperties({
+				values: [],
+				value: '',
+				placeHolder: loc.locationBrowser,
+				width: '400px'
+			}).component();
+
+			this.booksDropdown = this.view.modelBuilder.dropDown().withProperties({
+				values: this.bookNames,
+				value: '',
+				width: '400px'
+			}).component();
+
+			this.notebookTypeDropdown = this.view.modelBuilder.dropDown().withProperties({
+				values: ['-', BookTreeItemType.Notebook, BookTreeItemType.Markdown],
+				value: '',
+				width: '400px'
+			}).component();
+
+			const browseFolderButton = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+				ariaLabel: loc.browse,
+				iconPath: IconPathHelper.folder,
+				width: '18px',
+				height: '20px',
+			}).component();
+
+			browseFolderButton.onDidClick(async () => {
+				this.saveLocationInputBox.value = await this.selectFolder();
+			});
+
+			this.booksDropdown.onValueChanged(async () => {
+				const book = this.books.find(b => b.bookItems[0].title === getDropdownValue(this.booksDropdown));
```

Comment:
You could have the list of books be a map instead so you don't have to search for it